substratevm/mx.substratevm/mx_substratevm.py

```diff
@@ -1076,6 +1078,8 @@ def debuginfotest(args, config=None):
     parsed = parser.parse_args(masked_args)
     output_path = unmask(parsed.output_path)[0]
     build_only = parsed.build_only
+    if config is None:
```

Comment:
Re:
> This change is meant to reduce the time spent on building native-image

If this is what you want to achieve then this change will have the opposite effect.

It states that you do not want to use the default build (i.e. `config == None`) that got built as part of running `mx build` but instead you want a custom graalvm release to be used for your image building in `debuginfotest`.

Specifically using this PR will cause the following to happen in addition to what happens without it:
```
Building GraalVM for config GraalVMConfig(primary_suite_dir=None, dynamicimports=(), disable_libpolyglot=True, force_bash_launchers=True, skip_libraries=True, components=('Native Image',), exclude_components=(), native_images=('native-image',)) ...
```
That will build an additional graalvm release with a native-image image that will then be used to build your images.
Are you sure you want that?

Without adding this the image build for will be carried out with the default graalvm release built by the substrate suite.

Reply:
> That will build an additional graalvm release with a native-image image that will then be used to build your images.
> Are you sure you want that?

No, it looks like you are right.

My issue is the following. I build `native-image` with e.g.:
```
mx --primary-suite-path substratevm --no-jlinking --components='Native Image' build
```

`mx graalvm-home` gives:
```
$ mx --primary-suite-path substratevm --no-jlinking --components='Native Image' graalvm-home          
/tmp/graal/sdk/mxbuild/linux-amd64/GRAALVM_ADB1ABFC21_JAVA11/graalvm-adb1abfc21-java11-22.0.0-dev
```

`native-image --version` gives:
```
$ /tmp/graal/sdk/mxbuild/linux-amd64/GRAALVM_ADB1ABFC21_JAVA11/graalvm-adb1abfc21-java11-22.0.0-dev/bin/native-image --version
native-image dev CE (Java Version 11.0.12+5-jvmci-21.3-b03)
```

`mx native-image` however gives:
```
$ mx --primary-suite-path substratevm --no-jlinking --components='Native Image' native-image --version
Can not find /home/zakkak/code/graal/sdk/mxbuild/linux-amd64/GRAALVM_CE0C781D19_JAVA11/graalvm-ce0c781d19-java11-22.0.0-dev/bin/native-image
```

and `mx debuginfotest` gives:
```
$ mx --primary-suite-path substratevm --no-jlinking --components='Native Image' debuginfotest
Building GraalVM for config None ...
```

So it looks like `mx ... native-image` and `mx ... debuginfotest` are ignoring the configuration passed through command line arguments. I would expect them to respect the config and reuse the `native-image` I built on the first step. As a result the right solution would be to make those commands respect the configuration (like `mx ... graalvm-home` does), right?

Reply:
I think what you need is something similar to 8953b8868351 but for the `--components` option. This should make `_run_graalvm_cmd` able to recognize your previous `mx --primary-suite-path substratevm --no-jlinking --components='Native Image' build` as the default (None) config.

Reply:
This should now be resolved, thanks for pointing me to the right direction!